server/agoraApi.yaml

```
@@ -180,7 +180,490 @@ paths:
     get:
       tags:
         - Goals
-      summary: Manage Listing, creating, updating and deleting user goals
+      summary: Retrieves all available goals, provide limit and offset for pagnation, top limit 100.
+      parameters:
+        - in: query
+          name: limit
+          schema:
+            type: integer
+          description: The number of records to return, maximum 100
+        - in: query
+          name: offset
+          schema:
+            type: integer
+          description: The starting record number, defaults to 0
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Goal'
+          headers:
+            x-agora-message-title:
+              schema:
+                type: string
+              description: Title of Agora API message
+            x-agora-message-message:
+              schema:
+                type: string
+              description: Message of Agora API message
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/UnauthorizedError'
+        '403':
+          $ref: '#/components/responses/ForbiddenError'
+        '404':
+          $ref: '#/components/responses/NotFoundError'
+    post:
+      tags:
+        - Goals
+      summary: Save a new goal. If the name of the goal already exists the current goal will not be duplicated. If the ownedBy member is provided with an existing goal name the owner will be updated to the provided value.  Only the goal name is required.
```

We should try to make this more concise; what do you think of the following?

```suggestion
      summary: Checks database for the goal in requestBody by Id. If found and ownedBy flag set, updates existing goal. If not found, creates new goal.
```